File: uniffi_bindgen/src/bindings/kotlin/templates/Int8Helper.kt
internal object FfiConverterByte {
    fun lift(v: Byte): Byte {
        return v
    }

    fun read(buf: ByteBuffer): Byte {
        return buf.get()
    }

    fun lower(v: Byte): Byte {
        return v
    }

    fun write(v: Byte, buf: RustBufferBuilder) {
        buf.putByte(v)
    }
}

File: uniffi_bindgen/src/bindings/python/templates/Int8Helper.py
class FfiConverterInt8(FfiConverterPrimitive):
    @classmethod
    def lower(cls, value):
        if not -2**7 <= value < 2**7:
            raise ValueError("i8 requires {} <= value < {}".format(-2**7, 2**7))
        return super().lower(value)

    @staticmethod
    def read(buf):
        return buf.readI8()

    @staticmethod
    def write(value, buf):
        buf.writeI8(value)
